src/nix/daemon.cc

```diff
@@ -55,19 +55,16 @@ struct AuthorizationSettings : Config {
     Setting<Strings> trustedUsers{
         this, {"root"}, "trusted-users",
         R"(
-          A list of names of users (separated by whitespace) that have
-          additional rights when connecting to the Nix daemon, such as the
-          ability to specify additional binary caches, or to import unsigned
-          NARs. You can also specify groups by prefixing them with `@`; for
-          instance, `@wheel` means all users in the `wheel` group. The default
-          is `root`.
+          A list of user names, separated by whitespace.
+          These users will have additional rights when connecting to the Nix daemon, such as the ability to specify additional [substituters](#conf-substituters), or to import unsigned [NAR](@docroot@/glossary.md#gloss-nar)s.
+
+          You can also specify groups by prefixing names with `@`.
+          For instance, `@wheel` means all users in the `wheel` group.
 
           > **Warning**
           >
-          > Adding a user to `trusted-users` is essentially equivalent to
-          > giving that user root access to the system. For example, the user
-          > can set `sandbox-paths` and thereby obtain read access to
-          > directories that are otherwise inacessible to them.
+          > Adding a user to `trusted-users` is essentially equivalent to giving that user root access to the system.
+          > For example, the user can set [`sandbox-paths`](#conf-sandbox-paths) and thereby obtain read access to directories that are otherwise inacessible to them.
```

Comment:
```suggestion
          > For example, the user can set [`sandbox-paths`](#conf-sandbox-paths) and thereby obtain read access to directories that are otherwise inaccessible to them, or they can replace store path contents that are used in critical ways.
```

Typo, and make it more serious; not just reading.

Reply:
```suggestion
          > For example, the user can set [`sandbox-paths`](#conf-sandbox-paths) and thereby obtain read and write access to store paths.
```

Can be brief?

Reply:
The write vector does not use `sandbox-paths`, but we could simplify the text by removing the read attack, which seems weaker (but maybe isn't strictly weaker).

```suggestion
          > For example, the user can replace store path contents that are used in critical ways.
```

(I would understand a desire to also simplify "that are used", but I think it's important because privilege is not intrinsic to store paths. It really is about their use.)

Reply:
```suggestion
          > For example, the user can access or replace store path contents that are critical for system security.
```

I think I'm really stumbling over something "being used in critical ways".